test/source/specs/base/concurrency/task.js

```diff
@@ -539,6 +539,36 @@ describe('Data.Task', () => {
         ex.cancel();
         return ex.promise().catch(v => $ASSERT(Cancelled.hasInstance(v)));
       });
+
+      it('Deals with exception', async () => {
+        let Result = undefined;
+        try {
+          let ex = await Task.of('a')
+            .map((() => false ? Result.a : Result.b))
+            .run()
+            .promise();
+          // Should have thrown by then
+          $ASSERT(false)
+        } catch (e) {
+          if (e.message != 'false') $ASSERT(e instanceof TypeError)
```

Comment:
This should be testing if the error is an instance of AssertionError, so:

```js
const { AssertionError } = require('assert');
if (!(e instanceof AssertionError)) {
  $ASSERT(e instanceof TypeError);
}
```

Reply:
Hi! I tried to use `assert.throws()` which would have been clearer, but it wouldn't compile. I  simplified the test though now. Hopefully it is clearer.